commands/publish/index.js

```diff
@@ -714,6 +720,16 @@ class PublishCommand extends Command {
 
     // undefined defaults to "latest" OR whatever is in pkg.publishConfig.tag
   }
+
+  getPreDistTag(pkg) {
+    if (!this.options.preDistTag) {
+      return;
+    }
+    const isPrerelease = (semver.prerelease(pkg.version) || []).shift();
```

Comment:
This logic is lifted from [here](https://github.com/lerna/lerna/blob/4e3f8d8087e0e2e8e83c7ea79558c517203583c2/core/package-graph/index.js#L25), I'd rather not duplicate it.
```suggestion
    const isPrerelease = this.packageGraph.get(pkg.name).prereleaseId;
```

Reply:
Actually, we're working with a `Package`, not a `PackageGraphNode` - the version isn't up to date in the graph when we need to get the `preDistTag`, so have to determine `prereleaseId` from the `Package` instances in the mapper.

Considering that this logic was already duplicated [here](https://github.com/lerna/lerna/blob/4e3f8d8087e0e2e8e83c7ea79558c517203583c2/commands/version/index.js#L283), I'll just separate out the logic itself. It should probably be tested directly anyway, critical little bit of logic here.

Reply:
Fix: https://github.com/lerna/lerna/pull/1991/commits/70fb6301c660755c015a2c2795cd97272fdfd874